File: Backend/OguzlarBelediyesi.Domain/PageContent.cs
namespace OguzlarBelediyesi.Domain;

public sealed record PageContent(
    string Title,
    string Subtitle,
    IReadOnlyList<string> Paragraphs,
    string? ImageUrl = null,
    string? MapEmbedUrl = null,
    IReadOnlyList<ContactDetail>? ContactDetails = null);

public sealed record ContactDetail(string Label, string Value);

File: UI/src/app/models/page-content.ts
export interface ContactDetail {
    label: string;
    value: string;
}

export interface PageContent {
    id: string;
    key: string;
    title: string;
    subtitle: string;
    paragraphs: string[];
    imageUrl?: string;
    mapEmbedUrl?: string;
    contactDetails?: ContactDetail[];
}
